### PR TITLE
Add support for Spinach

### PR DESCRIPTION
**Pull request by ayufan**
This adds support for Spinach: https://github.com/codegram/spinach

This MR covers:
- a new SpinachAdapter,
- a rspec tests for SpinachAdapter,
- a `spinach_examples` to run test on Travis-CI,
- updates README.md with description how to use it.

### CODE REVIEW

**Comment by ArturT** [comment]
@ayufan Great work. Thank you!

**Comment by ayufan** [comment]
@ArturT Will you release a new version of gem? :)

**Comment by ArturT** [comment]
@ayufan Yes. I'm on it. I need to do a few small changes and I will let you know.

**Comment by ArturT** [comment]
@ayufan I've just released knapsack 1.11.0. :)

File: lib/knapsack/adapters/spinach_adapter.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
+require 'spinach'
```
Comment (ArturT) [inline lib/knapsack/adapters/spinach_adapter.rb]: @ayufan I found out this causes problems when someone don't use spinach. I did fix in knapsack 1.11.1
https://github.com/ArturT/knapsack/blob/master/CHANGELOG.md#1111

Comment (ayufan) [inline lib/knapsack/adapters/spinach_adapter.rb]: Thanks :) I appreciate it 👍

**Comment by ArturT** [comment]
@ayufan I added Spinach support also to knapsack_pro gem. Maybe you would like to try it, it's free? I'd love to hear your feedback. http://knapsackpro.com You have a big project so it might be interesting how the knapsack_pro will work for you.

**Pull request merged**